src/Stratis.Bitcoin/BlockPulling/BlockPuller.cs

```diff
@@ -1054,22 +1054,22 @@ private void AddComponentStats(StringBuilder statsBuilder)
                 statsBuilder.AppendLine($"Queueued downloads: {unassignedDownloads}");
             }
 
-            double averageBlockSizeKb = this.GetAverageBlockSizeBytes() / 1024.0;
+            double avgBlockSizeBytes = this.GetAverageBlockSizeBytes();
```

Comment:
@noescape00 I suspect that this might cause the same issue that I fixed this morning in the other PR? If this is 0 then we have arithmetic error again?

Reply:
Block size can't be less than the size of a header which is 80 bytes. Therefore this value will never go to 0 or negative (unless there is a big precision problem)

Reply:
💯